test/unit/math/test_ad_matvar.hpp

```diff
@@ -47,6 +49,20 @@ void expect_near_rel_matvar(const std::string& message, T1&& x, T2&& y,
   }
 }
 
+template <typename T1, typename T2,
+          require_all_std_vector_vt<is_std_vector, T1, T2>* = nullptr,
+          require_all_std_vector_st<is_var, T1, T2>* = nullptr>
```

Comment:
Should these be `is_var_matrix` to stop it accidentally getting called with `var<double>`?

Reply:
So we actually want this to be `is_var` in the `st` since the function could be like `f(std::vector<std::vector<var<Matrix>>>, var)`